lib/cursor/QueryCursor.js

```diff
@@ -480,7 +480,7 @@ function _waitForCursor(ctx, cb) {
  */
 
 function _create(ctx, doc, populatedIds, cb) {
-  const instance = helpers.createModel(ctx.query.model, doc, ctx.query._fields);
+  const instance = helpers.createModel(ctx.query.model, [doc], ctx.query._fields);
```

Comment:
Instead of making this change here, check what happens if you just make this change when `execPost()` is called. That's what triggers post hooks.

Also, one place that's always handy to look is the commit where we introduced post hooks for cursors, see #9345 and https://github.com/Automattic/mongoose/commit/4dad667a04c4d8c6bad918a5169203ed13eb5ce9 . That may provide some hints.